modules/apps/callbacks/ica_test.go

```diff
@@ -0,0 +1,247 @@
+		},
+		{
+			"success: dest callback",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s"}}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: dest callback with other json fields",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s"}, "something_else": {}}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: dest callback with malformed json",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s"}, malformed}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: dest callback with missing address",
+			`{"dest_callback": {"address": ""}}`,
+			"none",
+			true,
+		},
+		{
+			"success: source callback",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}}`, callbackAddr),
+			types.CallbackTriggerAcknowledgementPacket,
+			true,
+		},
+		{
+			"success: source callback with other json fields",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}, "something_else": {}}`, callbackAddr),
+			types.CallbackTriggerAcknowledgementPacket,
+			true,
+		},
+		{
+			"success: source callback with malformed json",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}, malformed}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: source callback with missing address",
+			`{"src_callback": {"address": ""}}`,
+			"none",
+			true,
+		},
+		{
+			"failure: dest callback with low gas (error)",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			"none",
+			false,
+		},
+		{
+			"failure: source callback with low gas (error)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			types.CallbackTriggerAcknowledgementPacket,
+			false,
+		},
+		{
+			"failure: dest callback with low gas (panic)",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s", "gas_limit": "350000"}}`, callbackAddr),
+			"none",
+			false,
+		},
+		{
+			"failure: source callback with low gas (panic)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "350000"}}`, callbackAddr),
+			types.CallbackTriggerAcknowledgementPacket,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			icaAddr := s.SetupICATest()
+
+			s.ExecuteICATx(icaAddr, tc.icaMemo, 1)
+			s.AssertHasExecutedExpectedCallback(tc.expCallback, tc.expSuccess)
+		})
+	}
+}
+
+func (s *CallbacksTestSuite) TestICATimeoutCallbacks() {
+	// ICA channels are closed after a timeout packet is executed
+	testCases := []struct {
+		name        string
+		icaMemo     string
+		expCallback types.CallbackTrigger
+		expSuccess  bool
+	}{
+		{
+			"success: transfer with no memo",
+			"",
+			"none",
+			true,
+		},
+		{
+			"success: dest callback",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s"}}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: source callback",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}}`, callbackAddr),
+			types.CallbackTriggerTimeoutPacket,
+			true,
+		},
+		{
+			"success: dest callback with low gas (error)",
```

Comment:
this will just behave the same as other `dest callback` cases above, right? 

The testcase I think is maybe a little misleading with the (error) in the name also - would be in favour to remove if possible to reduce num of tests to maintain if the behaviour is what I described in the other comment

Reply:
The error cases were removed. There is only panic now.